internal/controllers/machine/machine_controller.go

```diff
@@ -491,6 +493,94 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 	return ctrl.Result{}, nil
 }
 
+// onPodDeletionOrEvictionStarted returns a function for a  m *clusterv1.Machine that updates
+// the drain information at the machine's DrainingSucceededCondition when a pod eviction starts.
+func onPodDeletionOrEvictionStarted(ctx context.Context, m *clusterv1.Machine) func(pod *corev1.Pod, usingEviction bool) {
+	log := ctrl.LoggerFrom(ctx)
+	return func(pod *corev1.Pod, usingEviction bool) {
+		var verbStr string
+		if usingEviction {
+			verbStr = "eviction"
+			drainingStr := fmt.Sprintf("Eviction of pod %s started from node %s", pod.Name, m.Status.NodeRef.Name)
+			if conditions.Get(m, clusterv1.DrainingSucceededCondition) != nil {
+				if !strings.Contains(conditions.Get(m, clusterv1.DrainingSucceededCondition).Message, drainingStr) {
+					conditions.MarkFalse(m, clusterv1.DrainingSucceededCondition, clusterv1.DrainingReason, clusterv1.ConditionSeverityInfo, drainingStr)
+				}
+			} else {
+				conditions.MarkFalse(m, clusterv1.DrainingSucceededCondition, clusterv1.DrainingReason, clusterv1.ConditionSeverityInfo, drainingStr)
+			}
+		} else {
```

Comment:
We are never setting d.DisableEviction on the drain helper. This means it is always false => usingEviction will always be true. Given that we basically hard-code eviction I would say let's drop handling usingEviction = false (let's return an error in this case, so that we notice if the drain helper code ever changes from under us)

Same for onPodDeletionOrEvictionFinished